cairo_programs/cairo/hints/widemul128.cairo

```diff
@@ -0,0 +1,11 @@
+fn main() {
+    let lhs: felt = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    let rhs: felt = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+
+    let product = lhs * rhs;
+
+    let low = product & ((1 << 128) - 1);  
+    let high = product >> 128;             
+
+    return ();
```

Comment:
You're making a multiplication of two 128 bits, stored on two limbs `low` and `high` but you're not actually using the WideMul128 hint.

Reply:
You need to find a Corelib function that makes use of the `WideMul128` hint:
- Look at libfuncs defined in the cairo-lang-sierra-to-casm that uses the `WideMul128` hint in their implementation
By searching on the [cairo-lang-sierra-to-casm crate](https://github.com/search?q=repo%3Astarkware-libs%2Fcairo%20%22hint%20WideMul128%22&type=code) of the Cairo compiler,  you can see that it is used in the u256 divmod, u512 divmod by u256 and the u128 multiplication libfuncs
The easiest libfunc directly using this hint is the u128 multiplication: [build_u128_guarantee_mul](https://github.com/starkware-libs/cairo/blob/86b6b559ea15359ffaf35b3aa6474836179c57ce/crates/cairo-lang-sierra-to-casm/src/invocations/int/unsigned128.rs#L88-L109) which creates the u128_guarantee_mul
- [Search](https://github.com/search?q=repo%3Astarkware-libs%2Fcairo%20u128_guarantee_mul&type=code) the u128_guaranteed_mul function in the [Cairo corelib](https://github.com/starkware-libs/cairo/blob/86b6b559ea15359ffaf35b3aa6474836179c57ce/corelib/src/integer.cairo#L69-L87)

Now you need to write a Cairo program that uses the function `u128_guarantee_mul()`, look at the function signature and do appropriately, something like this can be enough, as the U128 type operators have been overloaded:

```cairo
fn main() {
    let _ = 0x123456_u128 * 0xFEDCBA_u128;
}
```

Reply:
When you're not sure about your program using a particular hint, compile it and look at the JSON compilation artifacts, go to the `hints` property and see if the wanted hint is part of it